addons/hw_drivers/controllers/driver.py

```diff
@@ -366,8 +367,9 @@ def run(self):
                         d.daemon = True
                         d.start()
                         iot_devices[path] = d
-                        self.send_alldevices()
                         break
+            if added or removed:
```

Comment:
it is not the same condition under which you instantiate a driver, you also check the presence/absence in iot_devices